### PR TITLE
Add DestinationRule for Citadel

### PR DESCRIPTION
**Pull request by johscheuer**
Reopened https://github.com/istio/istio/pull/19498 see https://github.com/istio/istio/issues/17886#issuecomment-565157293 for more information.

### CODE REVIEW

**Comment by johscheuer** [comment]
I think the misleading think here is that the `node_agent` and `citadel` still talking over (m)TLS at least wireshark tells this. Probably the certificate from citadel (istio_ca) is the issue since the SPIFFE URI is missing and the ingress-gw want to verificate the certificate of citadel (at leas tcpdump + wireshark shows this strange behaviour).

File: install/kubernetes/helm/istio/charts/security/templates/meshexpansion.yaml [hunk 1]
```diff
@@ -53,4 +53,27 @@ spec:
...
+    release: {{ .Release.Name }}
+    istio: citadel
+spec:
+  host: istio-citadel.{{ .Release.Namespace }}.svc.{{ .Values.global.proxy.clusterDomain }}
+  trafficPolicy:
+    portLevelSettings:
+    - port:
+        number: 8060
+      tls:
+        mode: DISABLE
```
Comment (irisdingbj) [inline install/kubernetes/helm/istio/charts/security/templates/meshexpansion.yaml]: Does this mean we disabled mTLS for citadel at all? Is there a way to make mTLS works for citadel?

Comment (johscheuer) [reply install/kubernetes/helm/istio/charts/security/templates/meshexpansion.yaml]: The funny thing is that the same DR is set for pilot -> https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/charts/pilot/templates/meshexpansion.yaml#L69-L87

**Comment by irisdingbj** [comment]
@vadimeisenbergibm Can you take a look at this PR? How this works in multi-cluster scenario?

**Pull request closed without merging**